### PR TITLE
Plugins: Forward user header (X-Grafana-User) in backend plugin requests

### LINKED ISSUE #47734: Datasource proxy headers broken since version 8.3
**What happened**:
HTTP Headers are not being passed to the datasource since version 8.3.0

**What you expected to happen**:
HTTP Headers to be passed to the datasource

**How to reproduce it (as minimally and precisely as possible)**:
Setup grafana 8.3.0 or higher behind a reverse proxy that injects some headers (or set dataproxy.send_user_header = true in config). Add a datasource and dump headers. Observe missing headers. Downgrade to 8.2.7 and headers appear again.

**Anything else we need to know?**:

**Environment**:
- Grafana version: 8.3.0 =<
- Data source type & version: Prometheus 2.32.1
- OS Grafana is installed on: Docker
- User OS & Browser: macOS & Safari
- Grafana plugins: none

https://github.com/grafana/grafana/issues/47734#tasklist-block-efb3e408-f8db-4b00-bd97-4e26bb10a91d

### PR DESCRIPTION
**Pull request by marefr**
**What is this feature?**
Grafana would pass X-Grafana-User header to backend plugins when [send_user_header](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#send_user_header)  is enabled.

**Why do we need this feature?**
This has been supported for the datasource and plugin proxies for long time. Since Grafana introduced support for backend plugins users are expecting/needs the user header to be forwarded for backend plugin requests.

**Who is this feature for?**
Plugin developers and end users relying on [send_user_header](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#send_user_header) configuration.

**Which issue(s) does this PR fix?**:
Fixes #47734
Dependent on #58132
Ref #60077

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/util/proxyutil/proxyutil_test.go [hunk 1]
```diff
@@ -77,3 +78,30 @@ func TestClearCookieHeader(t *testing.T) {
...
+		ApplyUserHeader(false, req, &user.SignedInUser{Login: "admin"})
+		require.NotContains(t, req.Header, "X-Grafana-User")
+	})
+
+	t.Run("Should not apply user header for anonomous user", func(t *testing.T) {
+		req, err := http.NewRequest(http.MethodGet, "/", nil)
+		require.NoError(t, err)
+
+		ApplyUserHeader(true, req, &user.SignedInUser{IsAnonymous: true})
+		require.NotContains(t, req.Header, "Cookie")
```
Comment (wbrowne) [inline pkg/util/proxyutil/proxyutil_test.go:97]: Should this not be
```suggestion
		require.NotContains(t, req.Header, "X-Grafana-User")
```

File: pkg/util/proxyutil/proxyutil.go [hunk 2]
```diff
@@ -61,3 +66,11 @@ func ClearCookieHeader(req *http.Request, keepCookiesNames []string, skipCookies
 func SetProxyResponseHeaders(header http.Header) {
 	header.Set("Content-Security-Policy", "sandbox")
 }
+
+// ApplyUserHeader Set the X-Grafana-User header if needed (and remove if not).
+func ApplyUserHeader(sendUserHeader bool, req *http.Request, user *user.SignedInUser) {
+	req.Header.Del(UserHeaderName)
+	if sendUserHeader && !user.IsAnonymous {
```
Comment (xnyo) [inline pkg/util/proxyutil/proxyutil.go:73]: Question: Is `user` guaranteed to be != `nil` when called from the `DataSourceProxy` and `PluginProxy` (`proxy.ctx.SignedInUser`)?

Comment (marefr) [reply pkg/util/proxyutil/proxyutil.go:73]: Good point we should probably check that. I'll fix

**Review by xnyo — APPROVED** [review]
Amazing ✨. LGTM!

**Pull request merged**